packages/exchange/src/exchange/exchange.py

```diff
@@ -16,6 +16,8 @@
 from exchange.token_usage_collector import _token_usage_collector
 from exchange.tool import Tool
 
+from goose.utils.confirm_execute import is_cancelled_by_user
```

Comment:
can't import goose in exchange 

I'd generally try to leave all of the changes for this mode out of the exchange code base! I'd contain it to just goose